test/sharness/t0182-circuit-relay.sh

```diff
@@ -50,7 +56,14 @@ test_expect_success 'connect B <-> Relay' '
 '
 
 test_expect_success 'wait until relay is ready to do work' '
-  sleep 1
+  while ! ipfsi 2 swarm connect /p2p/$PEERID_1/p2p-circuit/p2p/$PEERID_0; do
+    iptb stop &&
+    iptb_wait_stop &&
+    iptb start -wait -- --routing=none &&
+    iptb connect 0 1 &&
+    iptb connect 2 1 &&
+    sleep 5
+  done
```

Comment:
:information_source:  below is not a blocker,  imo this PR can be merged and if this is a real problem we will figure it out with  RC1 being used in the real world.

:warning:  Tests in  `go-ipfs/test/sharness/t0182-circuit-relay.sh`   became  flaky.

`ipfsi 2 swarm connect /p2p/$PEERID_1/p2p-circuit/p2p/$PEERID_0`   sometimes fails with:
```
NO_RESERVATION (204)
Error: connect 12D3KooWLuY1Q13o6Q91NoJn4Qv7RSq5rYS6uC4YCzKCJ3S8GnwQ failure: failed to dial 12D3KooWLuY1Q13o6Q91NoJn4Qv7RSq5rYS6uC4YCzKCJ3S8GnwQ:
  * [/p2p/12D3KooWNfs8uFpQf6NnseZKLyS3c8EFppGtUGejHpViBeg8f7Vt/p2p-circuit] error opening relay circuit: NO_RESERVATION (204)`
```

- it happens randomly, bumping it to 10 seconds does not help much, it still fails sometimes.
- shutting down nodes and starting them again fixes the problem
  - ...but it feels like covering up some underlying racy bug:   once `NO_RESERVATION` error is returned, the relay will never work, no matter how long we wait and retry again  (only way to fix it is to reboot of the relay node /  testbed). 



Repro (runs tests until they error):

```console
$ killall ipfs ; i=0; while ./t0182-circuit-relay.sh -v; do echo " -----> run no. $i <----"; sleep 1; ((i=i+1)) ; done
```

Reply:
cc @marten-seemann  for visibility – lmk if I should fill a bug in go-libp2p, or how I can provide more details

Reply:
Continued in https://github.com/ipfs/go-ipfs/pull/8967